integ/test_extended_counters.py

```diff
@@ -91,25 +155,49 @@ def wait_file(path, timeout=5):
 
 class TestInteg(object):
     def test_counters(self, servers):
-        "Tests adding kv pairs"
+        "Tests adding counters with (legacy behaviour)"
         server, _, output = servers
         server.sendall("foobar:100|c\n")
         server.sendall("foobar:200|c\n")
         server.sendall("foobar:300|c\n")
 
         wait_file(output)
         out = open(output).read()
-        assert "counts.foobar.count|600" in out
-        assert "counts.foobar.rate|600" in out
+        assert "counts.foobar.count|3" in out
+        assert "counts.foobar.rate|300" in out
 
     def test_counters_sample(self, servers):
-        "Tests adding kv pairs"
+        "Tests adding counters with sampling (legacy behaviour)"
         server, _, output = servers
         server.sendall("foobar:100|c|@0.1\n")
         server.sendall("foobar:200|c|@0.1\n")
         server.sendall("foobar:300|c|@0.1\n")
 
+        wait_file(output)
+        out = open(output).read()
+        assert "counts.foobar.count|30" in out
```

Comment:
This was actually wrong in the actual code, and the test was working because "3" is inside the "30" string. Took be a while to realize why this test was working when looking at the code